### PR TITLE
Use bound value in date filters

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Responds to feedback here https://github.com/PostHog/posthog/pull/7647#discussion_r767470127

## How did you test this code?

Update logic tests

### CODE REVIEW

**Review by mariusandra — DISMISSED** [review]
I guess the `defaultValue` comment was more to point out possible tech debt, not that this needs to be worked on now :). However if we're already here, I'll ask for some changes.

For a while I've been saying we need to disconnect all the various filters (e.g. `insightDateFilter`) from the URL and instead directly get the date range for the insight from `insightLogic`. This is what I think should happen here as well. Mainly because of this new line in `<InsightDateFilter />`:

```ts
const logic = insightDateFilterLogic({ dateFrom: props.dateFrom, dateTo: props.dateTo })
```

This leaves the illusion of a controlled component, but if we'd pass in new props on a new render, the date filter itself won't change. The props here could actually be written as:

```ts
const logic = insightDateFilterLogic({ defaultDateFrom: props.dateFrom, defaultDateTo: props.dateTo })
```

... since they are by very definition uncontrolled props now.

Thus, if you're already working on this, let's finish it up by removing `urlToAction` from insightLogic, and instead pass the data directly.

**Comment by neilkakkar** [comment]
To add more to this: Doing what Marius says would make Experimentation screens a lot nicer as well (the url is different, and we don't want the date params in there, when we can get them from the insight). We've been struggling a bit to make this date filter work properly because we're outside of the insights page and trying to use an insight filter to control it.

File: frontend/src/scenes/insights/InsightDateFilter/InsightDateFilter.tsx [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
 import { insightDateFilterLogic } from './insightDateFilterLogic'
 import { DateFilterProps, DateFilter } from 'lib/components/DateFilter/DateFilter'
 import './index.scss'
+import { insightLogic } from 'scenes/insights/insightLogic'
 
 export function InsightDateFilter(props: DateFilterProps): JSX.Element {
-    const logic = insightDateFilterLogic({ dateFrom: props.dateFrom, dateTo: props.dateTo })
+    const { insightProps } = useValues(insightLogic)
+    const { setFilters } = useActions(insightLogic)
+    const logic = insightDateFilterLogic({ dashboardItemId: undefined, ...insightProps })
```
Comment (mariusandra) [inline frontend/src/scenes/insights/InsightDateFilter/InsightDateFilter.tsx:11]: Why the extra `dashboardItemId: undefined` here?

**Review by mariusandra — COMMENTED** [review]
Hey, the changes are looking good. Some inline comments, and not sure what's up with the E2E tests.

File: frontend/src/scenes/insights/InsightDateFilter/InsightDateFilter.tsx (hunk 1 above)
Comment (alexkim205) [reply frontend/src/scenes/insights/InsightDateFilter/InsightDateFilter.tsx:11]: good catch, didn't mop up this leftover

**Comment by mariusandra** [comment]
Is this ready to review, or still a draft? Something's still up with the E2E tests as well.

**Comment by alexkim205** [comment]
I've been leaving this in draft until I figure out the e2e tests, but haven't had too much time to figure this out yet. I'll let you know asap once this is ready for review!

---

Update: ready to review now! @mariusandra 

The e2e tests actually caught a pretty nasty bug where setting the date filters were resetting the insight type to "Trends" every time. I abstracted `setFilters` into a `setDates` action in `insightDateFilterLogic` that takes care of this case.

**Comment by alexkim205** [comment]
This PR will be revisited once #7837 is merged in.

**Comment by mariusandra** [comment]
#7837 is finally in...

**Pull request closed without merging**